### ISSUE TITLE
Feature: Application CommandBar

### ISSUE THREAD

**Issue by mvegaca** [Microsoft/WindowsTemplateStudio#1099]
Related to #1098 and #425 a good idea would be to add a CommandBar feature.
This feature must:
- Add a command bar in app
- Handle position (top on PC and bottom on mobile)
- Allow pages and features specify commands to the bar (examples)
  - My position on Map page
  - Refresh on WebView page

**Comment by mrlacey** [comment]
I remember some discussion about adding this previously. The result of that discussion was that this is too broad an issue to be something we could add in a way that is widely usable. This should be discussed, planned, and defined very thoroughly before any work begins on this.

The two related issues (#425 & #1098) already show up one problem. Where should the bar go? One wants it always at the top and the other wants it always at the bottom. There's now a third option being discussed, that of having it at the top when running on a PC and at the bottom on a mobile device. There's no mention of other devices. If PC&mobile are being used for wide and narrow screens then is it ok for all apps to have the bar move as the screen is resized? 

Should the CommandBar be added to individual pages or across all pages?
- If individual pages, then which ones? 
Should all functionality that can be moved go into the command bar? 
What if people don't want it there? 
Should it become a new option for such pages? Would this need two versions of each page? 
I don't believe we should be adding new functionality to pages (e.g. my position on map page) just to warrant the inclusion of the command bar. Or is this specific example something to add to that page? - if so a separate issue is warranted.

- If a single bar that appears across all pages:
What differences need to be considered for NavigationPanel vs Tab&Pivots?
For NavPanel apps, if at the top of the screen should it appear above, below or beside the hamburger icon? How and should this change as the screen is resized and the drawer state changes?
For Tab&Pivots, should the bar appear above or inline with the headers? (like in the store app when maximized on a wide screen.) Or only inline if it doesn't crash into the tab header?
Should commands can be handled centrally or passed to the currently displayed page? Or both? And how is best to pass them?
If passing to the displayed page, does this mean that every page must be changed to potentially handle commands being passed?



These and many more questions need answering before this can be implemented.

**Comment by ralarcon** [comment]
I like the idea of having a powerfull command bar as feature. At the same time, every dev could add it's own implementation so this feature is not a limitation itself. I would suggest to setup a PoC to help address all considerations marked by @mrlacey as any other things may raise. 

The discussion is still open but moving this to milestone 1.5 as 1.4 is a short sprint...

**Comment by mvegaca** [comment]
I'm going to start a PoC on this

**Comment by mrlacey** [comment]
Just like any external contribution, the requirements should be defined here in the issue before doing any work.

**Comment by crutkas** [comment]
Guys, we have way higher priorities than this from a feature aspect
